### PR TITLE
Add storage to immutable accts

### LINKED ISSUE #17: Add method to increase storage capacity for immutable storage accounts
This method needs to support v1 & v2 storage accounts.

### PR DESCRIPTION
**Pull request by dboures**
closes #17 and #19

### CODE REVIEW

**Review by VegetarianOrc — APPROVED** [review]
Just a couple of small things, but LGTM!

File: example/src/bin/add_immutable_storage.rs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+async fn make_storage_immutable<T: Signer + Send + Sync>(
+    shdw_drive_client: &ShadowDriveClient<T>,
+    storage_account_key: &Pubkey,
+) {
+    let storage_account = shdw_drive_client
+        .get_storage_account(storage_account_key)
+        .await
+        .expect("failed to get storage account");
+    match storage_account {
```
[the remark this answers is not in the document]
Comment (dboures) [reply example/src/bin/add_immutable_storage.rs:69]: It does derive `Debug`, but I get something like this when I try to just print: 

`error[E0609]: no field `reserved_bytes` on type `StorageAcct`
  --> src/bin/add_immutable_storage.rs:95:48
   |
95 |     println!("old size: {:?}", storage_account.reserved_bytes);
   |                                                ^^^^^^^^^^^^^^`

Maybe we give them a trait in the future? I'm not sure what best practice is, I'm not so familiar with structs inside an Enum

File: src/client/add_immutable_storage.rs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    ) -> ShadowDriveResult<StorageResponse> {
+        let size_as_bytes: u64 = size
+            .get_bytes()
+            .try_into()
+            .map_err(|_| Error::InvalidStorage)?;
+
+        let wallet_pubkey = self.wallet.pubkey();
+        let (user_info, _) = derived_addresses::user_info(&wallet_pubkey);
+
+        let user_info_acct = self.rpc_client.get_account(&user_info).await;
```
[the remark this answers is not in the document]
Comment (dboures) [reply src/client/add_immutable_storage.rs]: Great catch

File: src/client/reduce_storage.rs [hunk 3]
```diff
@@ -78,7 +78,8 @@ where
             }
         };
 
-        self.send_shdw_txn("reduce-storage", txn_encoded).await
+        self.send_shdw_txn::<StorageResponse>("reduce-storage", txn_encoded)
```
[the remark this answers is not in the document]
Comment (dboures) [reply src/client/reduce_storage.rs]: I didn't know that was possible, very cool

**Pull request merged**